test/functional/ex_cmds/write_spec.lua

```diff
@@ -94,6 +97,16 @@ describe(':write', function()
     fifo:close()
   end)
 
+  it("creates missing parent directories", function()
+    command("write ++p test/write/p_opt.txt")
+    source([[
+      edit test/write/p_opt.txt
+      call setline(1, ['line1'])
+      write
+    ]])
+    eq(eval("['line1']"), eval("readfile('test/write/p_opt.txt')"))
```

Comment:
is it enough to just check filereadable()? (and check that it is NOT readable before the ++p step). Then we can skip the source() part.